### PR TITLE
Fixed usage of null coalescing operator when operands are UnityEngine.Object

### PR DESCRIPTION
**Pull request by LostPolygon**
It's better to avoid using null coalescing operator when operands are descendants of UnityEngine.Object. The reason is: null coalescing operator basically calls Object.ReferenceEquals(...), which only compares the references themselves, ignoring the operator overloads. UnityEngine.Object implements overload for != and == operators, as it is nothing more that a wrapper for C++ object, and it _is_ possible for a C# object to have no corresponding C++ object attached to it. This is what happened here. Little known fact, but serialized properties of Unity API types _always_ have a C# object attached, even when they look empty from the Inspector. Why? This is made to spit out error messages like ```"UnassignedReferenceException - The variable SomeTransform of 'Transform' has not been assigned. You probably need to assign the SomeTransform variable of the Transform script in the inspector"```. And Unity is not doing this does in the builds. So bam, we have a "real" null in the build, but in the Editor we have an actual object disguised as null.

TL;DR: this was leading to a different behaviour between playing from the Unity Editor, a playing from a build. This took *quite* some time to figure out...

### CODE REVIEW

**Comment by micahosborne** [comment]
Wow, nice work, that is something that would be so easy to over look! Thank you so much!
